packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -467,7 +467,7 @@ export function indexHtmlMiddleware(
           return next()
         }
         if (
-          fullBundleEnv.triggerBundleRegenerationIfStale() ||
+          (await fullBundleEnv.triggerBundleRegenerationIfStale()) ||
```

Comment:
What's the different compared to use `DevEngine#ensureLatestBuild`?

Reply:
`scheduleBuildIfStale` returns whether we need to wait for the build or not.